sw/device/tests/dif/dif_gpio_smoketest.c

```diff
@@ -0,0 +1,66 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/dif/dif_gpio.h"
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/check.h"
+#include "sw/device/lib/testing/test_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"  // Generated.
+
+static dif_gpio_t gpio;
+const test_config_t kTestConfig;
+
+// Pins 20-23 are grounded and pins 16-19 are reserved for JTAG.
+const uint32_t kGpioMask = 0xFF00FFFF;
```

Comment:
Wouldn't driving a signal still appear on the input even for the grounded pins ([relevant illustration](https://docs.opentitan.org/hw/ip/gpio/doc/gpio_blockdiagram.svg))?

Reply:
It wouldn't because AFAICT (looking at `hw/top_earlgrey/rtl/top_earlgrey_nexysvideo.sv` lines 79-80), `gpio_i[*]` signals are grounded. @imphil please correct me if I'm wrong.